backend/core/src/shared/email.service.ts

```diff
@@ -46,6 +86,33 @@ export class EmailService {
     )
   }
 
+  private partial(view: string) {
+    return fs.readFileSync(
+      path.join(path.resolve(__dirname, "..", "..", "views").replace("/dist", ""), `/${view}`),
+      "utf8"
+    )
+  }
+
+  private partials() {
+    const partials = {}
+    const dirName = path.resolve(__dirname, "..", "..", "views/partials").replace("/dist", "")
+
+    fs.readdirSync(dirName).forEach((filename) => {
+      partials[filename.slice(0, -4)] = this.partial("partials/" + filename)
+    })
+
+    return partials
+  }
+
+  private translations() {
+    return JSON.parse(
+      fs.readFileSync(
+        path.resolve(__dirname, "..", "..", "locals/general.json").replace("/dist", ""),
+        "utf8"
+      )
+    )
+  }
```

Comment:
It's probably a better practice to use the async versions of all these functions rather than blocking the whole app while they get read. This would then make this service an [async provider](https://docs.nestjs.com/fundamentals/async-providers), which _slightly_ complicates the declaration, but it's probably the "right" way to do it.

As long as this service is a [default scoped](https://docs.nestjs.com/fundamentals/injection-scopes#provider-scope) provider (basically a singleton loaded at initialization), it probably doesn't matter too much because the blocking code should only run once at initialization time. But if we ever wanted to re-scope this, it could definitely cause blocking issues.

Reply:
@bencpeters I tried using async version. It was working in development, but in test env sth was off with variable visibility and scoping and no partials were loaded. Is it ok if I add TODO for this one for future and leave it for now as we are under time preasure?

Reply:
Yes